src/compiler/checker.ts

```diff
@@ -11882,8 +11882,18 @@ namespace ts {
                 if (symbol.flags & SymbolFlags.Property &&
                     (expr.kind === SyntaxKind.PropertyAccessExpression || expr.kind === SyntaxKind.ElementAccessExpression) &&
                     (expr as PropertyAccessExpression | ElementAccessExpression).expression.kind === SyntaxKind.ThisKeyword) {
-                    const func = getContainingFunction(expr);
-                    return !(func && func.kind === SyntaxKind.Constructor && func.parent === symbol.valueDeclaration.parent);
+                    return !isInConstructor(getContainingFunction(expr));
+                    function isInConstructor(func: FunctionLikeDeclaration) {
+                        if (!func)
+                            return false;
+                        if (func.kind !== SyntaxKind.Constructor)
+                            return false;
+                        if (func.parent === symbol.valueDeclaration.parent) // If the symbol was declared in the class:

```

Comment:
In the comment. you mean the symbol has to be referred inside constructor of the class that defined it? (i.e it can't be subclass)